python/lsst/pipe/tasks/setPrimaryFlags.py

```diff
@@ -36,25 +36,31 @@ class SetPrimaryFlagsConfig(Config):
 class SetPrimaryFlagsTask(Task):
     ConfigClass = SetPrimaryFlagsConfig
 
-    def __init__(self, schema, **kwargs):
+    def __init__(self, schema, isSingleFrame=False, **kwargs):
         Task.__init__(self, **kwargs)
         self.schema = schema
-        self.isPatchInnerKey = self.schema.addField(
-            "detect_isPatchInner", type="Flag",
-            doc="true if source is in the inner region of a coadd patch",
-        )
-        self.isTractInnerKey = self.schema.addField(
-            "detect_isTractInner", type="Flag",
-            doc="true if source is in the inner region of a coadd tract",
-        )
+        self.isSingleFrame = isSingleFrame
+        if not self.isSingleFrame:
+            self.isPatchInnerKey = self.schema.addField(
+                "detect_isPatchInner", type="Flag",
+                doc="true if source is in the inner region of a coadd patch",
+            )
+            self.isTractInnerKey = self.schema.addField(
+                "detect_isTractInner", type="Flag",
+                doc="true if source is in the inner region of a coadd tract",
+            )
+            primaryDoc = ("true if source has no children and is in the inner region of a coadd patch "
+                          "and is in the inner region of a coadd tract "
+                          "and is not \"detected\" in a pseudo-filter (see config.pseudoFilterList)")
+        else:
+            primaryDoc = "true if source has no children and is not a sky source"
         self.isPrimaryKey = self.schema.addField(
             "detect_isPrimary", type="Flag",
-            doc="true if source has no children and is in the inner region of a coadd patch "
-                "and is in the inner region of a coadd tract "
-                "and is not \"detected\" in a pseudo-filter (see config.pseudoFilterList)",
+            doc=primaryDoc,
         )
 
-    def run(self, sources, skyMap, tractInfo, patchInfo, includeDeblend=True):
+    def run(self, sources, skyMap=None, tractInfo=None, patchInfo=None,
+            includeDeblend=True):
         """Set is-primary and related flags on sources
 
         @param[in,out] sources   a SourceTable
```

Comment:
Please take this opportunity to numpydoc this run method 🙂 I also suggest expanding the main doc string a bit to remind us what "is-primary" means and what any "related flags" may be.

Reply:
I've updated this docstring to below (please let me know if this looks okay):
        """Set is-patch-inner, is-tract-inner and is-primary flags on sources.
        For coadded imaging, the is-primary flag returns True when an object
        has no children, is in the inner region of a coadd patch, is in the
        inner region of a coadd trach, and is not detected in a pseudo-filter
        (e.g., a sky_object).
        For single frame imaging, the is-primary flag returns True when a
        source has no children and is not a sky source.

        Parameters
        ----------
        sources :
            A sourceTable. Reads in centroid fields and an nChild field.
            Writes is-patch-inner, is-tract-inner, and is-primary flags.
        skyMap : lsst.skymap.BaseSkyMap
            Sky tessellation object
        tractInfo : lsst.skymap.TractInfo
            Tract object
        patchInfo : lsst.skymap.PatchInfo
            Patch object
        includeDeblend : `bool`
            Include deblend information in isPrimary?
        """

Reply:
Thanks! Please do add whatever type `sources` is.

Reply:
Done, thanks!